### ISSUE TITLE
Lock management

### ISSUE THREAD

**Issue by mastercoms** [ProjectBorealis/PBSync#9]
PBSync can automatically set read-only for unlocked files, unlock unchanged assets, and lock changed ones, as Git LFS in conjunction with UE4 Ed can get stuck in invalid states sometimes.

**Comment by mastercoms** [comment]
Read-only set done in `7f4fb9e31e8b931e0626a0d3068c9741f7b859da`

**Comment by mastercoms** [comment]
Read only functionality done in 7f4fb9e31e8b931e0626a0d3068c9741f7b859da

**Comment by commodo** [comment]
does this require anything else to be closed?

**Comment by mastercoms** [comment]
Yes, it should unlock assets if they are up to date or behind all of the 3 main branches: `master`, `promoted`, and `trunk`.

**Comment by mastercoms** [comment]
Also it may be better to use the `git-lfs` executable within the UE4GitPlugin (`engine_dir/Engine/Plugins/ProjectBorealis/UE4GitPlugin/git-lfs.exe`), since that supports multi-threaded lock/unlock.
